modules/nf-lineage/src/main/nextflow/lineage/model/WorkflowOutputs.groovy

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2013-2025, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.lineage.model
+
+import groovy.transform.Canonical
+import groovy.transform.CompileStatic
+import nextflow.lineage.serde.LinSerializable
+
+import java.time.OffsetDateTime
+
+/**
+ * Models the results of a workflow execution.
+ *
+ * @author Jorge Ejarque <jorge.ejarque@seqera.io
+ */
+@Canonical
+@CompileStatic
+class WorkflowOutputs implements LinSerializable {
```

Comment:
Continuing the discussion here about making workflow runs Merkle-compliant:
1. Rename `WorkflowRun` to `WorkflowLaunch`
2. Rename this class to `WorkflowRun`
3. Rename the `workflowRun` field in this class to `launch`
4. Use the hash of this class as the "workflow run id" in the lineage log

This way, the `WorkflowRun` represents the "final" record that is created and the entrypoint into the lineage from the runs log. It resolves the issue where `lid://<hash>#outputs` requires a reverse lookup. Even if we never go full CID, it still makes sense to do it

Reply:
We could do a similar thing for tasks, i.e. `TaskRun` -> `TaskLaunch` and `TaskOutputs` -> `TaskRun`

Reply:
I'm still a bit hesitant about this change. I'd prefer to address some quick wins above, merge and keep this for next iteration/discussion

Reply:
Once we release a first version and people start playing with it, it's going to be harder to make changes like renaming entities. This is why I want to resolve this before 25.04

Reply:
No hurry, this is going to be a preview release